### ISSUE TITLE
[IMPORTANT] CODE STYLE STANDARD

### ISSUE THREAD

**Issue by imkiva** [covariant-institute/mozart#12]
Seeing #11, I found that we can no longer rely on external tools or IDE to help us ensure the unification of code style, so I think we should specify a code standard, according to which the code style will be checked in the future code review.

看到 #11 这个 PR，我发现我们不能再依赖外部工具或者 IDE 来帮助我们保证代码风格的统一，所以我认为我们应该指定一个代码标准，日后的 Code Review 就按照这份标准来检查代码风格。

The first problem to be solved is **braces**, here is my opinion:
首先要解决的问题是**大括号**，我的意见是这样：

```cpp
namespace NS {
    class FUCK {
    public:
        FUCK() {
            // do something
        }
    };
}
```

**Comment by mikecovlee** [comment]
I agree. But I think the braces of function should be placed at the next line.
Why not using astyle as formatting tools? CLion can't run in command line environment. Actually I don't use IDEs very often.

**Comment by mikecovlee** [comment]
My suggest is that using `astyle -A4 -N` as our official formatting options.

**Comment by mikecovlee** [comment]
Otherwise, pls speak modern Chinese instead of classical Chinese.

**Comment by imkiva** [comment]
astyle is not so wise as you can see my review comments in #11, which clion does better.

**Comment by imkiva** [comment]
> braces of function should be placed at the next line.

Why should we put the braces in different positions and increase coding complexity? It's easy to think that our braces are not standardized at all.

**Comment by mikecovlee** [comment]
Some additional proposal:
1. A virtual destructor is required in non-final classes/structures to avoid unnecessary memory leak.
2. Declaring all constructors explicitly or using `default` and `delete` if you don't want.

**Comment by imkiva** [comment]
Added 1 rule:

* Do not eliminate param name in function declaration.
